fauna/models.py

```diff
@@ -16,43 +17,130 @@ def __hash__(self):
         hash(self.name)
 
 
-class DocumentReference:
-    collection: Module
-    ref_id: str
+class BaseReference:
+    _collection: Module
 
-    def __init__(self, collection: Union[str, Module], ref_id: str):
-        if isinstance(collection, str):
-            self.collection = Module(collection)
+    def __init__(self, coll: Union[str, Module]):
+        if isinstance(coll, Module):
+            self._collection = coll
+        elif isinstance(coll, str):
+            self._collection = Module(coll)
         else:
-            self.collection = collection
+            raise TypeError(
+                f"Coll must be either a Module or a string, but was a {type(coll)}"
+            )
 
-        self.ref_id = ref_id
+    @property
+    def coll(self) -> Module:
+        return self._collection
+
+
+class DocumentReference(BaseReference):
+    _id: int
+
+    def __init__(self, coll: Union[str, Module], ref_id: int):
+        super().__init__(coll)
+        self._id = ref_id
```

Comment:
this should have the same `isinstance` check as `BaseReference`, and maybe cast strings to ints.

Reply:
I went back to storing strings, so added the isinstance check for string. I'm not going to do a trial conversion to int, though. It will leave ID validity up to Fauna